tests/integration_tests/functional/test_snapshot_basic.py

```diff
@@ -412,3 +412,42 @@ def test_create_large_diff_snapshot(test_microvm_with_api):
 
     # If the regression was not fixed, this would have failed. The Firecracker
     # process would have been taken down.
+
+
+def test_diff_snapshot_overlay(guest_kernel, rootfs, microvm_factory):
+    """
+    Tests that if we take a diff snapshot and direct firecracker to write it on
+    top of an existing full snapshot file, it will successfully merge them.
```

Comment:
is it not misleading as we take a diff snapshot first?

Reply:
Yeah, there's a comment why we have to do that further down in the test

Reply:
My concern is someone may not have time to read through the entire test and the description may give them a wrong impression. Although the test name may help with that, so take it as a nit.